xapian-letor/api/featurelist.cc

```diff
@@ -104,12 +114,13 @@ FeatureList::create_feature_vectors(const Xapian::MSet & mset,
 	Xapian::Document doc = i.get_document();
 	std::vector<double> fvals;
 	internal->set_data(letor_query, letor_db, doc);
+	Feature::Internal* internal_feature = new Feature::Internal(letor_db,
+								    letor_query,
+								    doc);
+	internal->populate_feature_internal(internal_feature);
 	for (Feature* it : internal->feature) {
-	    it->set_database(letor_db);
-	    it->set_query(letor_query);
-	    it->set_doc(doc);
+	    it->internal = internal_feature;
 	    // Computes and populates the Feature with required stats.
-	    internal->populate_feature(it);
```

Comment:
You've removed the code, but left the comment saying what it did.

Reply:
fixed in a fixup commit.